service/worker/deployment/deployment_workflow.go

```diff
@@ -25,20 +25,18 @@
 package deployment
 
 import (
-	"time"
+	"sync"
 
 	sdkclient "go.temporal.io/sdk/client"
 	sdklog "go.temporal.io/sdk/log"
-	"go.temporal.io/sdk/temporal"
 	"go.temporal.io/sdk/workflow"
 	deployspb "go.temporal.io/server/api/deployment/v1"
 )
 
 type (
 	DeploymentLocalState struct {
 		*deployspb.DeploymentWorkflowArgs
-		DeploymentName string
-		BuildID        string
+		lock sync.Mutex // preventing concurrent updates
```

Comment:
You don't need (and shouldn't have) locks in workflow code, it's forced to be single-threaded

Reply:
@dnr Is it the case for Update handlers too? What if an update handler is long and calls a bunch of activities? Should we use these new wf [synchronization primitives](https://temporal.io/blog/robust-message-handlers#synchronization-primitives) in that case?

Reply:
cc - @ShahabT 
I guess my initial assumption of concurrent updates to the workflow can thus be resolved; thanks David I wasn't sure about this until now

Reply:
You can use the new `workflow.Mutex` if you need to, but you definitely can't use `sync.Mutex`. Do we need it here? I don't think so since the update handler doesn't do any blocking calls

Reply:
I thought the update handler does blocking calls since they are sync in nature - Shahab and I were worried about the case: "What happens when we have concurrent update requests? Do we need to prevent our local state from this concurrency issue"?

Now, I had a hunch that a workflow processes concurrent requests sequentially but was not sure if we can apply that to this use-case

Reply:
Not right now, but in Shivam's next PR an activity will be added to the handler. `workflow.Mutex` can be added on that PR.

So as far as this PR is concerned, we know SDK will take care of synchronization of the update handlers so no two update handlers will be running concurrently (unless when they are blocked), right?

Reply:
Ah ok, if you're going to add an activity and need the activity to complete before another update runs, then I think workflow.Mutex would be appropriate. You can do it in this PR, just make sure it's using workflow.Mutex